Makefile

```diff
@@ -1,61 +1,117 @@
+keystore_path = $(wildcard ~)/keystores/kolibri
+apk_path = bin
 
-# Clear out apks
+build_tools_path = $(wildcard ~/.buildozer/android/platform/android-sdk-21/build-tools/22.0.1)
+
+# Specify where to look for keystores, apk
+vpath %.keystore $(keystore_path)
+vpath %.apk $(apk_path)
+
+
+# ENVIRONMENT SETUP TOOLS
+
+
+## Clear out apks
+.PHONY: clean
 clean:
-	# Make bin directory, where APK is held. Empty if it exists.
-	mkdir -p bin
-	rm -f bin/*.apk 2> /dev/null
+	mkdir -p $(apk_path)
+	rm -f $(apk_path)/*.apk 2> /dev/null
 	rm -rf ./src/kolibri 2> /dev/null
 
-# Update build system (download NDK/SDK, build Python, etc)
+## Update build system (download NDK/SDK, build Python, etc)
+.PHONY: updatedependencies
 updatedependencies:
 	buildozer android update
 
-# Replace the default loading page, so that it will be replaced with our own version
+## Replace the default loading page, so that it will be replaced with our own version
+.PHONY: replaceloadingpage
 replaceloadingpage:
 	rm -f .buildozer/android/platform/build/dists/kolibri/webview_includes/_load.html
 	cp ./assets/_load.html .buildozer/android/platform/build/dists/kolibri/webview_includes/
 	cp ./assets/loading-spinner.gif .buildozer/android/platform/build/dists/kolibri/webview_includes/
 
-# Extract the whl file
+## Extract the whl file
+.PHONY: extractkolibriwhl
 extractkolibriwhl:
 	unzip -q "src/kolibri*.whl" "kolibri/*" -x "kolibri/dist/cext*" -d src/
 
-# Generate the andoid version
+## Generate the andoid version
+.PHONY: generateversion
 generateversion:
 	python ./scripts/generateversion.py
 
-# Buld the debug version of the apk
-builddebugapk:
-	buildozer android debug
 
-# Buld the release version of the apk
-buildreleaseapk:
-	buildozer android release
+# APK SIGNING
 
 
-# DOCKER BUILD
+## Creates a keystore in user's home directory
```

Comment:
So, I think that all this stuff would live better in the main Kolibri repo. It's because all Docker logic lives there ATM and it just assumes that the installer repo lives in the right environment. So the installer repo can be assuming the same: That it will be provided the right repo for its scripts to run.

Meanwhile, that doesn't mean that EVERYTHING here should be dockerized, so we can perhaps look into some kind of simplification of the Makefile and whatever scripts its calling.

A good rule of thumb for a make target is: Someone should want to call it separately. If it's useless in a separate context, then it's better to normalize it into the callee.